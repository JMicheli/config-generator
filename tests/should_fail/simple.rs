//! Should fail because "env" isn't one of the attributes we define.

use config_generator::ConfigGenerator;

#[derive(ConfigGenerator, Default)]
struct TestStruct {
  #[env = "NAME_ENV_KEY"]
  pub name: String,
  pub description: Option<String>,
  pub x: u32,
  pub y: u32,
}

fn main() {
  let test_struct = TestStruct::default().with_env();
}
