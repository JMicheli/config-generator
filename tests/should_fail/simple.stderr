error: cannot find attribute `env` in this scope
 --> tests/should_fail/simple.rs:7:5
  |
7 |   #[env = "NAME_ENV_KEY"]
  |     ^^^
  |
  = note: `env` is in scope, but it is a function-like macro
